appdevguide/deviceSupport.md

```diff
@@ -0,0 +1,501 @@
+
+- Record support layers can also get notified about this process by making their address field special, in which case the record type's special routine can refuse to accept the new address before it is presented to the device support layer.
+Special cannot prevent the old device support from being disconnected however.
+
+
+If an address change is refused, the change to the INP or OUT field will cause an error or exception to be passed to the software performing the change.
+If this was a Channel Access client the result is to generate an exception callback.
+
+To switch to a different device support layer, it is necessary to change the DTYP field before the INP or OUT field.
+The change to the DTYP field has no effect until the latter field change takes place.
+
+If a record is set to I/O Interrupt scan but the new layer does not support this, the scan will be changed to Passive.
+
+### Initialization/Registration
+
+Device support that implements the extended behaviour must provide an `init` routine in the Device Support Entry Table 
+(see Section [Initialize Device Processing](#initialize-device-processing) ).
+In the first call to this routine (pass 0) it registers the address of its Device Support eXtension Table (DSXT) in a call to `devExtend`.
+
+The only exception to this registration requirement is when the device support uses a link type of CONSTANT.  In this 
+circumstance the system will automatically register an empty DSXT for that particular support layer (both the 
+`add_record` and `del_record` routines pointed to by this DSXT do nothing and return zero). This exception allows 
+existing soft channel device support layers to continue to work without requiring any modification, since the iocCore 
+software already takes care of changes to PV\_LINK addresses.
+
+The following is an example of a DSXT and the initialization routine that registers it:
+
+```C
+static struct dsxt myDsxt = {
+    add_record, del_record
+};
+
+static long init(int pass) {
+    if (pass==0) devExtend(&myDsxt);
+    return 0;
+}
+```
+
+A call to `devExtend` can only be made during the first pass of the device support initialization process, and registers the DSXT for that device support layer.
+If called at any other time it will log an error message and immediately return.
+
+### Device Support eXtension Table
+
+The full definition of {external+epics-base:cpp:class}`struct dsxt <dsxt>` is found in {doc}`epics-base:devSup_h`.
+
+It has at least the following members:
+
+- {external+epics-base:cpp:member}`dsxt::add_record`
+
+- {external+epics-base:cpp:member}`dsxt::del_record`
+
+
+There may be future additions to this table to support additional functionality; such extensions may only be made by changing the devSup.h header file and rebuilding EPICS Base and all support modules, thus neither record types nor device support are permitted to make any private use of this table.
```

Comment:
exceedingly long line

...and all support modules. Neither...
(split sentence; remove 'thus')